dht/token-sharding.hh

```diff
@@ -35,4 +35,56 @@ unsigned shard_of(unsigned shard_count, unsigned sharding_ignore_msb_bits, const
 
 token token_for_next_shard(const std::vector<uint64_t>& shard_start, unsigned shard_count, unsigned sharding_ignore_msb_bits, const token& t, shard_id shard, unsigned spans);
 
+class sharding_info {
+protected:
+    unsigned _shard_count;
+    unsigned _sharding_ignore_msb_bits;
+    std::vector<uint64_t> _shard_start;
+public:
+    sharding_info(unsigned shard_count = smp::count, unsigned sharding_ignore_msb_bits = 0);
+    /**
+     * Calculates the shard that handles a particular token.
+     */
+    virtual unsigned shard_of(const token& t) const;
```

Comment:
Why virtual?

Reply:
because there's dummy_sharding_info in tests that has its own logic for testing.

Reply:
Seems wasteful to make this heavily-called function virtual. Can't dummy_sharding_info use the standard logic?

I'll comment there when I reach that commit.